### PR TITLE
Change the FS submodule to async

### PR DESCRIPTION
**Pull request by liqinggd**

### CODE REVIEW

File: src/libos/crates/async-mountfs/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+        let inode = Self::new(
+            self.inner.create(name, type_, mode).await?,
+            self.vfs.clone(),
+        );
+        Ok(inode)
+    }
+
+    /// Strong type version of `find()`
+    pub async fn find(&self, name: &str) -> Result<Arc<Self>> {
+        let mut result = self.self_ref.upgrade().unwrap();
```
Comment (tatetian) [inline src/libos/crates/async-mountfs/src/lib.rs:167]: Choose more descriptive names, e.g., `arc_self` and `this`.

**Review by tatetian — COMMENTED** [review]
The code looks good to me in general. I raised some minor issues. Thanks.

File: src/libos/src/net/unix/trusted/trusted_addr.rs [hunk 2]
```diff
@@ -105,6 +105,20 @@ impl TrustedAddr {
...
     }
+
+    // Init inode field of TrustedAddr for connect.
+    pub async fn try_init_inode(&mut self) -> Result<()> {
+        if let UnixAddr::Pathname(path) = &self.unix_addr {
+            let inode_num = {
+                let current = current!();
+                let fs = current.fs();
+                let inode_file = fs.lookup_inode(&FsPath::try_from(path.as_ref())?).await?;
+                inode_file.metadata().await?.inode
```
[the remark this answers is not in the document]
Comment (liqinggd) [reply src/libos/src/net/unix/trusted/trusted_addr.rs:124]: No, cannot guarantee

File: src/libos/src/net/unix/trusted/trusted_addr.rs [hunk 3]
```diff
@@ -105,6 +105,20 @@ impl TrustedAddr {
...
         return None;
     }
+
+    // Init inode field of TrustedAddr for connect.
+    pub async fn try_init_inode(&mut self) -> Result<()> {
+        if let UnixAddr::Pathname(path) = &self.unix_addr {
+            let inode_num = {
+                let current = current!();
+                let fs = current.fs();
+                let inode_file = fs.lookup_inode(&FsPath::try_from(path.as_ref())?).await?;
```
[the remark this answers is not in the document]
Comment (liqinggd) [reply src/libos/src/net/unix/trusted/trusted_addr.rs:123]: I'm not clear why the file created in `bind_addr` is a normal file, if it is wrong, we can fix it in another commit. By the way, this commit just modify the API to async.

File: src/libos/crates/async-mountfs/src/lib.rs (hunk 1 above)
Comment (liqinggd) [reply src/libos/crates/async-mountfs/src/lib.rs]: Done. Thanks

File: etc/template/base.yaml [hunk 4]
```diff
@@ -13,3 +13,4 @@ targets:
       - root
       - sys
       - tmp
+      - async_sfs
```
Comment (volcano0dr) [inline etc/template/base.yaml:16]: I suggest changing the name to sfs.

Comment (liqinggd) [reply etc/template/base.yaml]: OK, thanks

**Review by volcano0dr — APPROVED** [review]
LGTM

**Pull request merged**